rumqttc/src/state.rs

```diff
@@ -72,13 +74,15 @@ pub struct MqttState {
     pub write: BytesMut,
     /// Indicates if acknowledgements should be send immediately
     pub manual_acks: bool,
+    /// Maximum outgoing packet size, set via MqttOptions
+    pub max_outgoing_packet_size: usize,
 }
 
 impl MqttState {
     /// Creates new mqtt state. Same state should be used during a
     /// connection for persistent sessions while new state should
     /// instantiated for clean sessions
-    pub fn new(max_inflight: u16, manual_acks: bool) -> Self {
+    pub fn new(max_inflight: u16, manual_acks: bool, max_outgoing_packet_size: usize) -> Self {
```

Comment:
we are exposing `MqttState`, so is also considered a breaking change. Its very unlikely that someone is using this but can you still mention this in CHANGELOG.